desktop/src/input.rs

```diff
@@ -0,0 +1,142 @@
+use ruffle_core::backend::input::InputBackend;
+use ruffle_core::events::KeyCode;
+use std::collections::HashSet;
+use winit::event::{ElementState, KeyboardInput, VirtualKeyCode};
+
+pub struct GlutinInputBackend {
+    keys_down: HashSet<VirtualKeyCode>,
+}
+
+impl GlutinInputBackend {
+    pub fn new() -> Self {
+        Self {
+            keys_down: HashSet::new(),
+        }
+    }
+
+    pub fn update(&mut self, input: KeyboardInput) {
+        match input.state {
+            ElementState::Pressed => {
+                if let Some(key) = input.virtual_keycode {
+                    self.keys_down.insert(key);
+                }
+            }
+            ElementState::Released => {
+                if let Some(key) = input.virtual_keycode {
+                    self.keys_down.remove(&key);
+                }
+            }
+        }
+    }
+}
+
+impl InputBackend for GlutinInputBackend {
+    fn is_key_down(&self, key: KeyCode) -> bool {
+        match key {
```

Comment:
Could we move the winit keycode -> Ruffle keycode mapping into a `to_ruffle_key_code` function? I get that it's not a 1-to-1 mapping, but maybe `keys_down` can change to `HashSet<KeyCode>`.

The reason is because this mapping will also be used elsewhere (for example, keyPress button events, which fire based on the OS key repeat rate).

Reply:
We can't keep track of `KeyCode` because it's not 1:1 - if you press lshift and rshift and let go of either, we'd lose track even though technically shift is still down. We have to track winit and convert on lookup. I can try to find a way to make this a reusable method though!